Tests/FunctionalTests/helpermethod.ts

```diff
@@ -139,6 +139,29 @@ export async function ClickWaitRetry(page: Page, buttonToClick: string, selector
   }
 }
 
+export async function ClickWaitRetryWarnings(page: Page, buttonToClick: string, selectorToWaitFor: string, warningVal: string,
+  timeout: number = 30000, step: number = 1000){
+const maxtime = Date.now() + timeout;
+let success = false;
+
+while (Date.now() < maxtime && !success)
+{
+await page.click(buttonToClick);
+if (await page.locator(selectorToWaitFor).textContent()==warningVal)
+{
+  success = true;
+}
+else{
+  await page.waitForSelector(selectorToWaitFor, {timeout: step});
```

Comment:
I think `page.waitForSelector` will throw an exception and should be surrounded by a try catch.

However, I'm not sure if the `ClickWaitRetryWarnings` is actually needed - instead could you have used `ClickWaitRetry` followed by `await page.locator(selectorToWaitFor).textContent()===warningVal` in the test?

If you decide to keep `ClickWaitRetryWarnings`, could you please fix the indentation and consider using `ClickWaitRetry` within it to avoid the duplication with it

Reply:
I have removed the `ClickWaitRetryWarnings()` method and handle in the test. Thanks for your suggestion.